tests/src/test/java/org/wso2/micro/gateway/tests/services/MutualSSLTestCase.java

```diff
@@ -0,0 +1,210 @@
+        String configPath = getClass().getClassLoader()
+                .getResource("confs" + File.separator + "mutualSSL-test.conf").getPath();
+        String[] args = {"--config", configPath};
+        System.out.println("MTSL TEST CASE");
+        microGWServer.startMicroGwServer(balPath, args);
+    }
+
+
+    @Test(description = "mutual SSL is properly established with ballerina keystore and trust store")
+    public void mutualSSLEstablished() throws Exception {
+
+        String trustStorePath = getClass().getClassLoader()
+                .getResource("keyStores" + File.separator + "ballerinaTruststore.p12").getPath();
+        Properties systemProps = System.getProperties();
+        systemProps.put("javax.net.debug", "ssl");
+        systemProps.put("javax.net.ssl.trustStore", trustStorePath);
+        systemProps.put("javax.net.ssl.trustStorePassword", "ballerina");
+        System.setProperties(systemProps);
+
+        SSLContext sslcontext;
+        KeyStore keyStore;
+        final char[] P12_PASSWORD = "ballerina".toCharArray();
+        final char[] KEY_PASSWORD = "ballerina".toCharArray();
+
+        try {
+
+            String keyPath = getClass().getClassLoader()
+                    .getResource("keyStores" + File.separator + "ballerinaKeystore.p12").getPath();
+            final InputStream is = new FileInputStream(keyPath);
+            keyStore = KeyStore.getInstance("pkcs12");
+            keyStore.load(is, P12_PASSWORD);
+            final KeyManagerFactory kmf = KeyManagerFactory.getInstance(KeyManagerFactory.getDefaultAlgorithm());
+            kmf.init(keyStore, KEY_PASSWORD);
+
+            sslcontext = SSLContext.getInstance("TLS");
+            sslcontext.init(kmf.getKeyManagers(), null, new java.security.SecureRandom());
+        } catch (Exception ex) {
+            throw new IllegalStateException("Failure initializing default SSL context", ex);
+        }
+
+        SSLSocketFactory sslsocketfactory = sslcontext.getSocketFactory();
+        try {
+
+            URL url = new URL("https://localhost:9595/pizzashack/1.0.0/menu");
+            HttpsURLConnection conn = (HttpsURLConnection) url.openConnection();
+            conn.setSSLSocketFactory(sslsocketfactory);
+            InputStream inputstream = conn.getInputStream();
+            System.out.println("Test is working properly");
+            InputStreamReader inputstreamreader = new InputStreamReader(inputstream);
+            BufferedReader bufferedreader = new BufferedReader(inputstreamreader);
+            String example = bufferedreader.readLine();
+            System.out.println(example);
+
+            while (example != null) {
+                System.out.println("Received " + example);
+                break;
+            }
+
+        } catch (UnknownHostException e) {
+            // TODO Auto-generated catch block
+            Logger logger = null;
+            logger.info("An UnknownHostException  occurred: " + e);
```

Comment:
Can we please make this a error log? Please refer https://github.com/wso2/carbon-apimgt/blob/6.x/components/apimgt/org.wso2.carbon.apimgt.hostobjects/src/main/java/org/wso2/carbon/apimgt/hostobjects/APIProviderHostObject.java#L145 to see how to add a logger for a particular class. 
Refer https://github.com/wso2/carbon-apimgt/blob/6.x/components/apimgt/org.wso2.carbon.apimgt.hostobjects/src/main/java/org/wso2/carbon/apimgt/hostobjects/APIProviderHostObject.java#L243 for exception logging.